### DISCUSSION TITLE
Onewire Temperature Sensor

### DISCUSSION THREAD

**Discussion by Jellybeanz1980** [UtilitechAS/amsreader-firmware#624] [General]
Hello,

I've connected a DS18B20 to IO pin 23, and selected this pin in the configuration page. Pin 1 and 3 of the temperature sensor has been connected to GND, and a pull-up (3.3V) resistor connected to pin 2 of the sensor.. The temperature sensor doesn't seem to be detected, nor found. Is there any small detail that I miss in order to activate the onewire interface?

Best regards,
Flemming

**Comment by ArnieO** [comment]
I don't recommend "Parasitic mode" powering - which is what you are trying. 
Please try powering it from 3,3V in stead (which means Vdd = Vpu in this diagram):
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/b21b0073-72fa-4198-9798-58987d2c5a03)

**Comment by Jellybeanz1980** [comment]
Thank you for the answer, Arnie. I use "parasitic power" configuration both with bit-banging, through transistor level-converting and onewire bridge from Dallas in various circuits without any issues. I'll try your suggestion.

I've placed debug output in updateTemperatures() in HwTools.cpp, which is the only reference to onewire, but this function doesn't seem to be called at all. Do I need to activate/include any functions, or is the onewire activated/included by default, and only the correct pin need to be chosen? Is there any particular pin(s) for the onewire on the ESP32?

**Reply by ArnieO** [reply]
OK, so you are building the code yourself? Not sure why your debug printout does not work then, we need to ask @gskjold.

All that should be needed is to select the right pin. In your case (GPIO23), this is what you need to do: 
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/a7cebeef-3d05-4406-9de3-2cef975f88e4)

**Reply by gskjold** [reply]
It should update the sensors every time the device receives data, or every 30s if not.
Regarding wiring, there are some links here: https://github.com/UtilitechAS/amsreader-firmware/wiki/Temperature-sensors which I have updated to point to the right place [LINK](https://www.milesburton.com/w/index.php/Dallas_Temperature_Control_Library)

As @ArnieO said, parasitic mode is not recommended, and the only needed configuration should be the GPIO setting.

**Comment by Jellybeanz1980** [comment]
Thank you for the information. I've solved the issue. I've found out that the temperature is updated in relation with reading the HAN data. As I was testing without a HAN connection, the temperature was not read.

Now I have tried both with parasitic mode and supplied mode, both configurations work without any issues.
